### ISSUE TITLE
Cannot compile std on rust 1.19.0

### ISSUE THREAD

**Issue by christianpoveda** [thepowersgang/mrustc#34]
Hello! I'm trying to compile rust following the instructions on the Readme.  When running
```
make -f minicargo.mk
```
I'm getting the following error:
```
BUILDING std from std v0.0.0 with features []
Calling /home/christian/Workspace/mrustc/bin/mrustc rustc-1.19.0-src/src/libstd/lib.rs --crate-name std --crate-type rlib -g -O -o output/libstd.hir -L output -l dl -l rt -l pthread --extern alloc=output/liballoc.hir --extern alloc_system=output/liballoc_system.hir --extern panic_abort=output/libpanic_abort.hir --extern collections=output/libcollections.hir --extern core=output/libcore.hir --extern libc=output/liblibc.hir --extern rand=output/librand.hir --extern compiler_builtins=output/libcompiler_builtins.hir --extern std_unicode=output/libstd_unicode.hir --extern unwind=output/libunwind.hir --extern rustc_asan=output/librustc_asan.hir --extern rustc_lsan=output/librustc_lsan.hir --extern rustc_msan=output/librustc_msan.hir --extern rustc_tsan=output/librustc_tsan.hir
DEBUG: Environment { OUT_DIR=/home/christian/Workspace/mrustc/output/build_std CARGO_MANIFEST_DIR=/home/christian/Workspace/mrustc/rustc-1.19.0-src/src/libstd CARGO_PKG_VERSION=0.0.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=0 }
:511: warning0:Unknown cfg() param 'target_vendor'
:511: warning0:Unknown cfg() param 'target_vendor'
rustc-1.19.0-src/src/libstd/sys/unix/net.rs:94: warning0:Unknown cfg() param 'target_vendor'
output/libstd.hir.o.c:49611:9: error: ‘_ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops5FnMut$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall_mut’ undeclared here (not in a function); did you mean ‘_ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops2Fn$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall’?
  (void*)_ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops5FnMut$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall_mut
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         _ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops2Fn$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall
C Compiler failed to execute
DEBUG: Compiler was terminated with signal 6
DEBUG: See output/libstd.hir_dbg.txt for the compiler output
BUILD FAILED
make: *** [minicargo.mk:42: output/libstd.hir] Error 1
```
I've also tried to run the following command, ending with the same output:
```
make -f minicargo.mk RUSTC_CHANNEL=stable RUSTC_VERSION=1.19.0 OVERRIDE_SUFFIX=-linux
```

**Comment by thepowersgang** [comment]
This is the same issue as seen on travis-ci, which I can't appear to reproduce locally. Is there anything particularly odd about your setup (GCC version maybe?)

If you're willing to go digging yourself, run `make -f minicargo.mk MRUSTC_DEBUG="Trans Enumerate:Trans Codegen"` to enable debug output (which may help locate the cause of this error)

**Comment by christianpoveda** [comment]
I have gcc 7.2.0. After running `make -f minicargo.mk MRUSTC_DEBUG="Trans Enumerate:Trans Codegen"` I got:

```
BUILDING std from std v0.0.0 with features []
Calling /home/christian/Workspace/mrustc/bin/mrustc rustc-1.19.0-src/src/libstd/lib.rs --crate-name std --crate-type rlib -g -O -o output/libstd.hir -L output -l dl -l rt -l pthread --extern alloc=output/liballoc.hir --extern alloc_system=output/liballoc_system.hir --extern panic_abort=output/libpanic_abort.hir --extern collections=output/libcollections.hir --extern core=output/libcore.hir --extern libc=output/liblibc.hir --extern rand=output/librand.hir --extern compiler_builtins=output/libcompiler_builtins.hir --extern std_unicode=output/libstd_unicode.hir --extern unwind=output/libunwind.hir --extern rustc_asan=output/librustc_asan.hir --extern rustc_lsan=output/librustc_lsan.hir --extern rustc_msan=output/librustc_msan.hir --extern rustc_tsan=output/librustc_tsan.hir
DEBUG: Environment { OUT_DIR=/home/christian/Workspace/mrustc/output/build_std CARGO_MANIFEST_DIR=/home/christian/Workspace/mrustc/rustc-1.19.0-src/src/libstd CARGO_PKG_VERSION=0.0.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=0 }
:511: warning0:Unknown cfg() param 'target_vendor'
:511: warning0:Unknown cfg() param 'target_vendor'
rustc-1.19.0-src/src/libstd/sys/unix/net.rs:94: warning0:Unknown cfg() param 'target_vendor'
output/libstd.hir.o.c:49611:9: error: ‘_ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops5FnMut$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall_mut’ undeclared here (not in a function); did you mean ‘_ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops2Fn$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall’?
  (void*)_ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops5FnMut$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall_mut
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         _ZRK$aLfn_1_$Rs__ZN3std9panicking9PanicInfo_$T0_as__ZN4core3ops2Fn$aL$T1_$Rs__ZN3std9panicking9PanicInfo$aR$aRcall
C Compiler failed to execute
DEBUG: Compiler was terminated with signal 6
DEBUG: See output/libstd.hir_dbg.txt for the compiler output
BUILD FAILED
make: *** [minicargo.mk:42: output/libstd.hir] Error 1
```

**Comment by thepowersgang** [comment]
See `output/libstd.hir_dbg.txt` for the compiler output - and it will be a wall of text (However, if you can provide the output in a gist, it'd be nice)

**Comment by christianpoveda** [comment]
Its arount 600Mb, i don't know what are the limits of gist but I cannot copy it right now

**Comment by thepowersgang** [comment]
Can you confirm the gcc version you're using (`g++ --version` in the same shell you compiled mrustc in)? I just compiled gcc 7.2.0 from source and encountered compilation errors.

**Comment by thepowersgang** [comment]
Correction, confirmed with fresh-compiled gcc 7.2.0.

**Comment by thepowersgang** [comment]
Fixed in 5f10912ba53c05b96a73675c7160f502c3075cb9
